pkg/network/devices.go

```diff
@@ -0,0 +1,51 @@
+package network
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+)
+
+// GetFreeDeviceName returns a device name that is unused; used when no network
+// name is provided
+func GetFreeDeviceName() (string, error) {
+	var (
+		deviceNum  uint
+		deviceName string
+	)
+	networkNames, err := GetNetworkNamesFromFileSystem()
+	if err != nil {
+		return "", err
+	}
+	liveNetworksNames, err := GetLiveNetworkNames()
+	if err != nil {
+		return "", err
+	}
+	for {
+		deviceName = fmt.Sprintf("%s-%d", CNIDeviceName, deviceNum)
+		logrus.Debugf("checking if device name %s exists in other cni networks", deviceName)
+		if used := StringInSliceContains(deviceName, networkNames); used {
+			deviceNum++
+			continue
+		}
+		logrus.Debugf("checking if device name %s exists in live networks", deviceName)
+		if used := StringInSliceContains(deviceName, liveNetworksNames); !used {
```

Comment:
No need for used variable

if StringInSliceContains(deviceName, liveNetworksNames) {
...
}